python/ray/data/_internal/pipelined_dataset_iterator.py

```diff
@@ -49,6 +50,12 @@ def iter_batches(
             _collate_fn=_collate_fn,
         )
 
+    def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]:
```

Comment:
Should we use `Mapping[str, Any]` instead of `TableRow`? I remember one UX study participant got tripped up and didn't realize `TableRow` works like a `dict`.

We could also do `dict[str, Any]`, but that's misleading because `TableRow` doesn't implement `__setitem__`.

Reply:
Let's discuss changes in separate PRs. This is just copying the comment.

Reply:
+1. This is just copying the existing API in `dataset.iter_rows`